modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -4611,6 +4630,61 @@ public IgniteSnapshotFutureImpl(IgniteInternalFuture<Void> fut) {
         }
     }
 
+    /** Snapshot check operation status. */
+    public class CheckOperationStatus implements AutoCloseable {
+        /** Snapshot metadata. */
+        private final SnapshotMetadata meta;
+
+        /** Operation request ID. */
+        private final UUID reqId;
+
+        /** Start time. */
+        private final long startTime = U.currentTimeMillis();
+
+        /** Processed partitions. */
+        private final AtomicInteger processedParts = new AtomicInteger();
+
+        /** Total partitions. */
+        private final Integer totalParts;
+
+        /** */
+        CheckOperationStatus(SnapshotMetadata meta, Integer totalParts, UUID reqId) {
+            this.meta = meta;
+            this.totalParts = totalParts;
+            this.reqId = reqId;
+        }
+
+        /** @return Snapshot metadata. */
+        public SnapshotMetadata metadata() {
+            return meta;
+        }
+
+        /** @return Operation request ID. */
+        public UUID requestId() {
+            return reqId;
+        }
+
+        /** @return Start time. */
+        public long startTime() {
+            return startTime;
+        }
+
+        /** @return Processed partitions. */
+        public AtomicInteger processedPartitions() {
```

Comment:
It should return `int`. Internal fields must not be exposed for modification. Let's add modification method instead.

Reply:
Fixed